src/main/java/org/myrobotlab/service/ServoMixer.java

```diff
@@ -175,305 +256,345 @@ public Pose loadPose(String name) {
+    if (p == null) {
+      error("cannot find pose %s", name);
+    }
+    moveToPose(name, p, false);
   }
 
-  public void removeSequence(String name) {
+  public void moveToPose(String name, Pose p, boolean blocking) {
     try {
-      if (!checkDir(servoMixerDirectory)) {
-        error("invalid poses directory %s", servoMixerDirectory);
+      
+      if (p.moves == null) {
+        error("no moves within pose file %s", name);
         return;
       }
-      String filename = new File(servoMixerDirectory).getAbsolutePath() + File.separator + name + ".seq";
-      File del = new File(filename);
-      if (del.exists()) {
-        del.delete();
+      
+      for (String sc : p.moves.keySet()) {
+        PoseMove pm = p.moves.get(sc);
+        ServoControl servo = (ServoControl) Runtime.getService(sc);
+        if (servo == null) {
+          warn("servo (%s) cannot move to pose because it does not exist", sc);
+          continue;
+        }
+        Double speed = pm.speed;
+        Double position = pm.position;
+        servo.setSpeed(speed);
+        if (blocking) {
+          servo.moveToBlocking(position);
+        } else {
+          servo.moveTo(position);
+        }
       }
-      invoke("getSequenceFiles");
+      invoke("publishStopPose", name);
     } catch (Exception e) {
       error(e);
     }
   }
 
-  public Sequence getSequence(String name) {
-    Sequence pose = null;
-    try {
-      if (!checkDir(servoMixerDirectory)) {
-        error("invalid poses directory %s", servoMixerDirectory);
-        return null;
-      }
-      String filename = new File(servoMixerDirectory).getAbsolutePath() + File.separator + name + ".seq";
-      log.info("Loading Pose name {}", filename);
-      pose = Sequence.loadSequence(filename);
-      broadcastState();
-    } catch (Exception e) {
-      error(e);
-    }
-    return pose;
+  @Override
+  public void onCreated(String name) {
+  }
+
+  @Override
+  public void onRegistered(Registration registration) {
   }
 
   /**
-   * Export Python representation of a sequence
-   * 
-   * @param name
-   *          does nothing..
-   * @return null
-   * 
+   * Part of service life cycle - a servo has been removed from the system
    */
-  public String exportSequence(String name) {
-    return null;
+  @Override
+  public void onReleased(String name) {
+    allServos.remove(name);
   }
 
-  public class Player implements Runnable {
-    Thread thread = null;
-    Sequence runningSeq = null;
-    int seqCnt = 0;
-    boolean running = false;
-
-    @Override
-    public void run() {
-      try {
-        running = true;
-        if (runningSeq.cycle) {
-          while (running) {
-            play();
-          }
-        } else {
-          play();
-        }
-      } catch (Exception e) {
-      }
-      running = false;
-    }
+  /**
+   * handles incoming selected events and publishes
+   * search events
+   * @param selected
+   */
+  public void onSelected(String selected) {

```

Comment:
Not really sure what this does, something to do with the GUI?

Reply:
sort of .. it doesn't currently set any state on the service, but it is an important publishing receiver for servo mixer
I created a SelectListener interface to more firmly tie things up ..

Reply:
![selectlistener](https://github.com/MyRobotLab/myrobotlab/assets/1266144/51c9ca5e-e358-4fec-9642-c360cd33ab57)